packages/features/ee/teams/lib/payments.ts

```diff
@@ -95,8 +102,22 @@ export const updateQuantitySubscriptionFromStripe = async (teamId: number) => {
     if (!url) return;
     const team = await getTeamWithPaymentMetadata(teamId);
     const { subscriptionId, subscriptionItemId } = team.metadata;
+    const membershipCount = team.members.length;
+    const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+    const subscriptionQuantity = subscription.items.data.find(
+      (sub) => sub.id === subscriptionItemId
+    )?.quantity;
+    if (!subscriptionQuantity) throw new Error("Subscription not found");
+
+    if (membershipCount < subscriptionQuantity) {
+      console.info(`Team ${teamId} has less members than seats, skipping updating subscription.`);
+      return;
+    }
```

Comment:
If an org has less than 30 members and adds more, we don't want to charge them for more members.